pkg/apis/configs/dataresidency/dataresidency_defaults.go

```diff
@@ -50,10 +51,8 @@ func NewDefaultsConfigFromMap(data map[string]string) (*Defaults, error) {
 
 	// Parse out the data residency configuration.
 	value, present := data[defaulterKey]
-	// Data residency configuration is not required, in which case it will mean
-	// allow all regions, so we simply use an empty one
 	if !present || value == "" {
-		return nc, nil
+		return nil, fmt.Errorf("ConfigMap is missing (or empty) key: %q : %v", defaulterKey, data)
```

Comment:
Are we sure we want this to be an error?

My main concern was typos and the `DisallowUnknownFields()` below should remove that possibility, right?

Reply:
Not really. The configmap has a weird way of parsing the file, so this is literally only checking if the key `default-dataresidency-config` exists. The function `parseEntry` actually only parse the `value` part only, which starts from `clusterDefaults`. So this conditions is triggered only when the `default-dataresidency-config` key is not present, same as the `gcp-auth` behavior.